### PR TITLE
fix: Improve the code related to on-demand-featureview.

### LINKED ISSUE #4202: Improve the code quality of on_demand_feature_view.py and its related transform functions.
## Expected Behavior 
The code quality should be high for Feast.

## Current Behavior
The SonarCloud service shows there are several improvements need to be done.

## Steps to reproduce
Analyze the code by  SonarCloudservice (free for open source project).
### Specifications

- Version: 0.37.1
- Platform:
- Subsystem:

## Possible Solution
Improve the code quality by following the suggestions from SonarCloud

### PR DESCRIPTION
**Pull request by shuchu**
# What this PR does / why we need it:
The SonarCloud (company which provides SonarCube) suggests several improvements about the code quality. 

Also, this PR is about to improve the code quality and the first step of improving running efficiency of "on-demand feature view" feature.

# Which issue(s) this PR fixes:

# Fixes #4202

### CODE REVIEW

File: sdk/python/feast/transformation/substrait_transformation.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
 from types import FunctionType
-from typing import Any, Dict, List
+from typing import Any
 
 import dill
 import pandas as pd
 import pyarrow
-import pyarrow.substrait as substrait  # type: ignore # noqa
```
Comment (tokoko) [inline sdk/python/feast/transformation/substrait_transformation.py]: @shuchu I know it looks like it's not used anywhere, but this is a required import :)

Comment (shuchu) [reply sdk/python/feast/transformation/substrait_transformation.py]: just learned from the unit-testing failures at the same time....   have added it back.

File: sdk/python/feast/transformation/python_transformation.py [hunk 2]
```diff
@@ -26,27 +26,19 @@ def __init__(self, udf: FunctionType, udf_string: str = ""):
...
-                f"input_dict should be type Dict[str, Any] but got {type(input_dict).__name__}"
-            )
+    def transform(self, input_dict: dict) -> dict:
         # Ensuring that the inputs are included as well
         output_dict = self.udf.__call__(input_dict)
-        if not isinstance(output_dict, Dict):
-            raise TypeError(
-                f"output_dict should be type Dict[str, Any] but got {type(output_dict).__name__}"
-            )
         return {**input_dict, **output_dict}
```
Comment (franciscojavierarceo) [inline sdk/python/feast/transformation/python_transformation.py:38]: Was this missing the return signature?

**Review by franciscojavierarceo — APPROVED** [review]
Lgtm!

File: sdk/python/feast/on_demand_feature_view.py [hunk 3]
```diff
@@ -62,24 +62,24 @@ class OnDemandFeatureView(BaseFeatureView):
     """
 
     name: str
-    features: List[Field]
```
Comment (sudohainguyen) [inline sdk/python/feast/on_demand_feature_view.py:65]: these types of change will break environment using py39, only support from py310

**Pull request merged**